### PR TITLE
implement strrchr

### PR DESCRIPTION
**Pull request by jrraymond**

### CODE REVIEW

File: src/string/src/lib.rs [hunk 1]
```diff
@@ -255,8 +255,17 @@ pub extern "C" fn strpbrk(s1: *const c_char, s2: *const c_char) -> *mut c_char {
 }
 
 #[no_mangle]
-pub extern "C" fn strrchr(s: *const c_char, c: c_int) -> *mut c_char {
-    unimplemented!();
+pub unsafe extern "C" fn strrchr(s: *const c_char, c: c_int) -> *mut c_char {
+    let len = strlen(s) as isize;
+    let c = c as i8;
+    let mut i = len - 1;
+    while *s.offset(i) != 0 && i >= 0 {
```
Comment (dlrobertson) [inline src/string/src/lib.rs]: Just a random question: What is the purpose of checking for the null character?

Comment (jrraymond) [reply src/string/src/lib.rs]: Good point. I think it’s uneceasary.

Comment (dlrobertson) [reply src/string/src/lib.rs]: :+1: thanks for working on this!

File: src/string/src/lib.rs [hunk 2]
```diff
@@ -259,7 +259,7 @@ pub unsafe extern "C" fn strrchr(s: *const c_char, c: c_int) -> *mut c_char {
     let len = strlen(s) as isize;
     let c = c as i8;
     let mut i = len - 1;
-    while *s.offset(i) != 0 && i >= 0 {
+    while i >= 0 {
```
Comment (Tommoa) [inline src/string/src/lib.rs]: Would `for i in (0..len).rev() { ... }` be more readable?

**Pull request merged**